browser-extension/mv2/src/shared/utils.js

```diff
@@ -4,6 +4,11 @@ RQ.Utils = RQ.Utils || {};
 RQ.Utils.regexFormat = "^/(.+)/(|i|g|ig|gi)$";
 
 RQ.Utils.isValidRegex = function (regexStr) {
+  try {
+    new RegExp(regexStr);
+  } catch {
+    return false;
+  }
```

Comment:
User input for regex also contains enclosing slashes and flags whereas `RegExp` constructor expects 2 arguments - string between slashes and flags.

If `isValidRegex` does not have other usages, you may combine it with `RQ.Utils.toRegex`.

```js
RQ.Utils.toRegex = function (regexStr) {
  const regexMatch = regexStr.match(new RegExp(RQ.Utils.regexFormat));
  if (regexMatch) {
    try {
      return new RegExp(regexMatch[1], regexMatch[2]);
    } catch (e) {
    }
  }
  return null;
};
```

Also, similar regex validation is at 2 other places:
`browser-extension/mv2/src/client/js/ruleMatcher.js`
and 
`browser-extension/mv3/src/service-worker/services/ruleMatcher.ts`.